### PR TITLE
[Localization] Add Japanese Localization

### PR DESCRIPTION
**Pull request by Tempo-anon**
## What are the changes?
Mirror of https://github.com/pagefaultgames/pokerogue/pull/2771

## Why am I doing these changes?
Add Japanese Localization

## What did change?
* Added Galmuri11.subset.wolff2 Galmuri9.subset.woff2, PokePT_Wansung.woff2, unifont-15.1.05.subset.woff2
* Added ja type icons
* Added Japanese localization

## How to test the changes?
Pull and change language

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by EnochG1** [comment]
Can you separate font between Chinese and Japanese?
Maybe Japanese font is applied to Chinese, too.
image

Original Chinese font
image

In your ja branch
image

**Comment by EnochG1** [comment]
Also, Japanese font sizes are inconsistent.
(For example in settings, I thought only the text in the selected setting was large, but it's not consistent.)
image

**Comment by EnochG1** [comment]
I added missed things https://github.com/Tempo-anon/pokerogue/pull/10/files

File: src/plugins/i18n.ts [hunk 1]
```diff
@@ -8,53 +8,65 @@ import { esConfig } from "#app/locales/es/config.js";
...
+    only: [ "zh" ],
+  },
+  // japanese
+  {
+    face: new FontFace("emerald", "url(./fonts/Galmuri11.subset.woff2)", { unicodeRange: rangesByLanguage.japanese }),
+    extraOptions: { sizeAdjust: "66%" },
+  },
+  {
+    face: new FontFace("pkmnems", "url(./fonts/Galmuri9.subset.woff2)", { unicodeRange: rangesByLanguage.japanese }),
+  },
```
Comment (EnochG1) [inline src/plugins/i18n.ts:61]:
```suggestion
  {
    face: new FontFace("emerald", "url(./fonts/Galmuri11.subset.woff2)", { unicodeRange: rangesByLanguage.japanese }),
    extraOptions: { sizeAdjust: "66%" },
    only: [ "ja" ],
  },
  {
    face: new FontFace("pkmnems", "url(./fonts/Galmuri9.subset.woff2)", { unicodeRange: rangesByLanguage.japanese }),
    only: [ "ja" ],
  },
```

File: src/plugins/i18n.ts [hunk 2]
```diff
@@ -8,53 +8,67 @@ import { esConfig } from "#app/locales/es/config.js";
...
+  // unicode (chinese)
+  {
+    face: new FontFace("emerald", "url(./fonts/unifont-15.1.05.subset.woff2)", { unicodeRange: rangesByLanguage.chinese }),
+    extraOptions: { sizeAdjust: "70%", format: "woff2" },
+    only: [ "zh" ],
+  },
+  {
+    face: new FontFace("pkmnems", "url(./fonts/unifont-15.1.05.subset.woff2)", { unicodeRange: rangesByLanguage.chinese }),
+    extraOptions: { format: "woff2" },
+    only: [ "zh" ],
```
Comment (EnochG1) [inline src/plugins/i18n.ts:52]:
```suggestion
    face: new FontFace("emerald", "url(./fonts/unifont-15.1.05.subset.woff2)", { unicodeRange: rangesByLanguage.chinese }),
    extraOptions: { sizeAdjust: "70%", format: "woff2" },
    only: [ "en", "es", "fr", "it", "de", "zh", "pt", "ko" ],
  },
  {
    face: new FontFace("pkmnems", "url(./fonts/unifont-15.1.05.subset.woff2)", { unicodeRange: rangesByLanguage.chinese }),
    extraOptions: { format: "woff2" },
    only: [ "en", "es", "fr", "it", "de", "zh", "pt", "ko" ],
```

**Review by EnochG1 — COMMENTED** [review]
To fix this BIG FONT in other language.

image

**Review by EnochG1 — COMMENTED** [review]
Maybe I found something

File: src/locales/ja/modifier-type.ts [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+    "ENEMY_ATTACK_BURN_CHANCE": { name: "やけどトークン" },
+    "ENEMY_STATUS_EFFECT_HEAL_CHANCE": { name: "なおしトークン", description: "まいターン2.5%のかくりつでじょうたいじょうしょうをかいふくする" },
+    "ENEMY_ENDURE_CHANCE": { name: "こらえるトークン" },
+    "ENEMY_FUSED_CHANCE": { name: "フュージョントークン", description: "やせいのポケモンがフュージョンするかくりつを1%ふやす" },
+  },
+  SpeciesBoosterItem: {
+    "LIGHT_BALL": { name: "Light Ball", description: "It's a mysterious orb that boosts Pikachu's Attack and Sp. Atk stats." },
+    "THICK_CLUB": { name: "Thick Club", description: "This hard bone of unknown origin boosts Cubone or Marowak's Attack stat." },
+    "METAL_POWDER": { name: "Metal Powder", description: "Extremely fine yet hard, this odd powder boosts Ditto's Defense stat." },
+    "QUICK_POWDER": { name: "Quick Powder", description: "Extremely fine yet hard, this odd powder boosts Ditto's Speed stat." }
```
Comment (xsn34kzx) [inline src/locales/ja/modifier-type.ts:253]:
```suggestion
    "LIGHT_BALL": { name: "でんきだま", description: "ピカチュウに 持たせると 攻撃と 特攻が あがる 不思議な玉。" },
    "THICK_CLUB": { name: "ふといホネ", description: "なにかの 硬いホネ。カラカラ または ガラガラに 持たせると 攻撃が あがる。" },
    "METAL_POWDER": { name: "メタルパウダー", description: "メタモンに 持たせると 防御が あがる 不思議な粉。とても こまかくて 硬い。" },
    "QUICK_POWDER": { name: "スピードパウダー", description: "メタモンに 持たせると 素早さが あがる 不思議 粉。とても こまかくて 硬い。" }
```
Localization for the `SpeciesBoosterItem`s using the same source from their initial PR (https://github.com/pagefaultgames/pokerogue/pull/1899#issuecomment-2168838329).

File: src/locales/ja/modifier-type.ts [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+
+    "LUCKY_EGG": { name: "しあわせタマゴ" },
+    "GOLDEN_EGG": { name: "おうごんタマゴ" },
+
+    "SOOTHE_BELL": { name: "やすらぎのすず" },
+
+    "SCOPE_LENS": { name: "Scope Lens", description: "It's a lens for scoping out weak points. It boosts the holder's critical-hit ratio."},
+    "LEEK": { name: "Leek", description: "This very long and stiff stalk of leek boosts the critical-hit ratio of Farfetch'd's moves."},
+
+    "EVIOLITE": { name: "Eviolite", description: "This mysterious evolutionary lump boosts the Defense and Sp. Def stats when held by a Pokémon that can still evolve." },
```
Comment (xsn34kzx) [inline src/locales/ja/modifier-type.ts:190]:
```suggestion
    "EVIOLITE": { name: "しんかのきせき", description: "進化の不思議な かたまり。持たせると 進化前ポケモンの 防御と 特防が あがる。" },
```
Localization for `EVIOLITE` using the same source from its initial PR (https://github.com/pagefaultgames/pokerogue/pull/2161#issuecomment-2164572537).

File: src/locales/ja/modifier-type.ts [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+    "SUPER_EXP_CHARM": { name: "いいけいけんちおまもり" },
+    "GOLDEN_EXP_CHARM": { name: "ゴールドけいけんちおまもり" },
+
+    "LUCKY_EGG": { name: "しあわせタマゴ" },
+    "GOLDEN_EGG": { name: "おうごんタマゴ" },
+
+    "SOOTHE_BELL": { name: "やすらぎのすず" },
+
+    "SCOPE_LENS": { name: "Scope Lens", description: "It's a lens for scoping out weak points. It boosts the holder's critical-hit ratio."},
+    "LEEK": { name: "Leek", description: "This very long and stiff stalk of leek boosts the critical-hit ratio of Farfetch'd's moves."},
```
Comment (xsn34kzx) [inline src/locales/ja/modifier-type.ts:188]:
```suggestion
    "SCOPE_LENS": { name: "ピントレンズ", description: "弱点が 見える レンズ。持たせた ポケモンの技が 急所に 当たりやすくなる。"},
    "LEEK": { name: "ながねぎ", description: "とても長くて 硬いクキ。カモネギに 持たせると 技が 急所に 当たりやすくなる。"},
```
Localization for `LEEK` and `SCOPE_LENS` using the same source from their initial PR (#2666).

**Review by xsn34kzx — COMMENTED** [review]
Filled in entries for items I've been responsible for with Scarlet/Violet's ripped text (from https://abcboy101.github.io/poke-corpus, more direct links attached for each suggestion). Definitely want another set of eyes on these for obvious reasons, but this should be good to start with.

**Comment by EnochG1** [comment]
I submitted PR for translating move, ability.
https://github.com/Tempo-anon/pokerogue/pull/11

**Comment by Chapybara-jp** [comment]
Hi! I PR'd several changes to #1861 which Chepyu accepted after it was first mirrored. It seems @chepyu had not actually stopped worked on that branch, so I'm not sure why it was closed. Either way, I can PR the changes I made to this branch as well.

**Comment by EnochG1** [comment]
> Hi! I PR'd several changes to #1861 which Chepyu accepted after it was first mirrored. It seems @Chepyu had not actually stopped working on that branch, so I'm not sure why it was closed. Either way, I can PR the changes I made to this branch as well.

@Chapybara-jp Hi,
Most branches were broken after 2nd 'beta' branch was created. (Also merge policy was changed since #1861) And we are trying to revive Japanese Localization.
It would be sooo good if you and @Chepyu give some help to Translate.
FYI, please feel free to join Discord :) either main(including dev channel) or Wiki(including i  game translation channel).

**Comment by Chapybara-jp** [comment]
> > Hi! I PR'd several changes to #1861 which Chepyu accepted after it was first mirrored. It seems @Chepyu had not actually stopped working on that branch, so I'm not sure why it was closed. Either way, I can PR the changes I made to this branch as well.
> 
> @Chapybara-jp Hi, Most branches(PR) were broken after 2nd 'beta' branch was created. (Also merge policy was changed after #1861) And we are trying to revive Japanese Localization. It would be sooo good if you and @Chepyu(or others) give some help to Translate. FYI, please feel free to join Discord :) either main(including dev channel) or Wiki(including i game translation channel).

Ah, I see! Then I'll PR my changes to this branch later today. I'll join the discord and Wiki as well!

**Comment by EnochG1** [comment]
@Chapybara-jp Perfect!
Please ping me when you join pokerogue discord. Because wiki(translation) discord is currently not public :)

File: src/locales/ja/move-trigger.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  "addType": "{{typeName}} was added to\n{{pokemonName}}!",
+  "cannotUseMove": "{{pokemonName}} cannot use {{moveName}}!",
+  "healHp": "{{pokemonName}} had its HP restored.",
+  "sacrificialFullRestore": "{{pokemonName}}'s Healing Wish\nwas granted!",
+  "invertStats": "{{pokemonName}}'s stat changes\nwere all reversed!",
+  "resetStats": "{{pokemonName}}'s stat changes\nwere eliminated!",
+  "faintCountdown": "{{pokemonName}}\nwill faint in {{turnCount}} turns.",
+  "copyType": "{{pokemonName}}'s type became the same as\n{{targetPokemonName}}'s type!",
+  "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
+  "swapArenaTags": "{{pokemonName}} swapped the battle effects affecting each side of the field!",
```
Comment (protimita) [inline src/locales/ja/move-trigger.ts:61]:
```suggestion
  "hitWithRecoil" : "{{pokemonName}}は\nはんどうによる　ダメージを　うけた！",
  "cutHpPowerUpMove": "{{pokemonName}}は\nたいりょくを　けずって　パワーぜんかい！",
  "absorbedElectricity": "{{pokemonName}}は\n　でんきを　きゅうしゅうした！",
  "switchedStatChanges": "{{pokemonName}}は　あいてと　じぶんのn\のうりょくへんかを　いれかえた！",
  "goingAllOutForAttack": "{{pokemonName}}は\nほんきを　だした！",
  "regainedHealth": "{{pokemonName}}は\nたいりょくを　かいふくした！",
  "keptGoingAndCrashed": "いきおいあまって　{{pokemonName}}は\nじめんに　ぶつかった！",
  "fled": "{{pokemonName}}は　にげだした!",
  "cannotBeSwitchedOut": "{{pokemonName}}を\nもどすことが　できない！",
  "swappedAbilitiesWithTarget": "{{pokemonName}}は\nおたがいの　とくせいを　いれかえた！",
  "coinsScatteredEverywhere": "こばんが　あたりに　ちらばった！",
  "attackedByItem": "{{pokemonName}}に\n{{itemName}}が　おそいかかる！",
  "whippedUpAWhirlwind": "{{pokemonName}}の　まわりで\nくうきが　うずをまく！",
  "flewUpHigh": "{{pokemonName}}は\nそらたかく　とびあがった！",
  "tookInSunlight": "{{pokemonName}}は\nひかりを　きゅうしゅうした！",
  "dugAHole": "{{pokemonName}}は\nじめんに　もぐった！",
  "loweredItsHead": "{{pokemonName}}は\nくびを　ひっこめた！",
  "isGlowing": "{{pokemonName}}を\nはげしいひかりが　つつむ！",
  "bellChimed": "すずのおとが　ひびきわたった！",
  "foresawAnAttack": "{{pokemonName}}は\nみらいに　こうげきを　よちした！",
  "hidUnderwater": "{{pokemonName}}は\nすいちゅうに　みをひそめた！",
  "soothingAromaWaftedThroughArea": "ここちよい　かおりが　ひろがった！",
  "sprangUp": "{{pokemonName}}は\nたかく　とびはねた！",
  "choseDoomDesireAsDestiny": "{{pokemonName}}は\nはめつのねがいを　みらいに　たくした！",
  "vanishedInstantly": "{{pokemonName}}の　すがたが\nいっしゅんにして　きえた！",
  "tookTargetIntoSky": "{{pokemonName}}は {{targetName}}を\nじょうくうに　つれさった！",
  "becameCloakedInFreezingLight": "{{pokemonName}}は\nつめたいひかりに　つつまれた！",
  "becameCloakedInFreezingAir": "{{pokemonName}}は\nこごえるくうきに　つつまれた！",
  "isChargingPower": "{{pokemonName}}は\nパワーを　ためこんでいる！",
  "burnedItselfOut": "{{pokemonName}}の　ほのうは\nもえつきた！",
  "startedHeatingUpBeak": "{{pokemonName}}は\nクチバシを　かねつしはじめた！",
  "isOverflowingWithSpacePower": "{{pokemonName}}に\nうちゅうの ちからが　あふれだす！",
  "usedUpAllElectricity": "{{pokemonName}}は\nでんきを　つかいきった！",
  "stoleItem": "{{pokemonName}}は\n{{targetName}}の {{itemName}}を　ぬすんだ！",
  "incineratedItem": "{{pokemonName}}は\n{{targetName}}の {{itemName}}を　もやした！",
  "knockedOffItem": "{{pokemonName}}は\n{{targetName}}の {{itemName}}を　はたきおとした！",
  "tookMoveAttack": "{{pokemonName}}は\n{{moveName}}の　こうげきを　うけた！",
  "cutOwnHpAndMaximizedStat": "{{pokemonName}}は\nたいりょくを　けずって　{{statName}}ぜんかい！",
  "copiedStatChanges": "{{pokemonName}}は　{{targetName}}の\nのうりょくへんかを　コピーした！",
  "magnitudeMessage": "マグニチュード{{magnitude}}！",
  "tookAimAtTarget": "{{pokemonName}}は {{targetName}}に\nねらいを　さだめた！",
  "transformedIntoType": "{{pokemonName}}は\n{{typeName}}タイプに　なった！",
  "copiedMove": "{{pokemonName}}は\n{{moveName}}を　コピーした！",
  "sketchedMove": "{{pokemonName}}は\n{{moveName}}を　スケッチした！",
  "acquiredAbility": "{{pokemonName}}の　とくせいが\n{{abilityName}}に　なった！",
  "copiedTargetAbility": "{{pokemonName}}は\n{{targetName}}の　{{abilityName}}を　コピーした！",
  "transformedIntoTarget": "{{pokemonName}}は\n{{targetName}}に　へんしんした！",
  "tryingToTakeFoeDown": "{{pokemonName}}は　あいてを\nみちづれに　しようとしている！",
  "addType": "{{pokemonName}}に\n{{typeName}}タイプが　ついかされた！",
  "cannotUseMove": "{{pokemonName}}は\n{{moveName}}を　つかえなかった！",
  "healHp": "{{pokemonName}}の\nたいりょくが　かいふくした！",
  "sacrificialFullRestore": "{{pokemonName}}の\nねがいごとが　かなった！",
  "invertStats": "{{pokemonName}}の\nのうりょくへんかが　ぎゃくてんした！",
  "resetStats": "{{pokemonName}}の\nのうりょくへんかが　もとにもどった！",
  "faintCountdown": "{{pokemonName}}は\n{{turnCount}}ターンごに　ほろびてしまう！",
  "copyType": "{{pokemonName}}は　{{targetPokemonName}}と\n同じタイプに　なった！",
  "suppressAbilities": "{{pokemonName}}の　とくせいが　きかなくなった！",
  "swapArenaTags": "{{pokemonName}}は\nおたがいの　ばのこうかを　いれかえた！",
```

**Review by protimita — COMMENTED** [review]
move-triggers.ts Japanese Translation.
All spaces are full-width.
regainHealth, healHP, and tookMoveAttack need in-game checking to see if they're used correctly.
acquiredAbility is improvised text, not original.

**Comment by EnochG1** [comment]
@protimita Due this PR is already merged so I would create new one and apply your suggestion. Please wait 🙏

**Pull request merged**